lib/src/main/java/uk/gov/di/ipv/core/library/service/CiStorageService.java

```diff
@@ -100,25 +132,186 @@ public void submitMitigatingVcList(
                         .withFunctionName(
-                                configService.getEnvironmentVariable(CI_STORAGE_GET_LAMBDA_ARN))
+                                configService.getEnvironmentVariable(
+                                        cimitGetContraindicatorsLambdaArn))
                         .withPayload(
                                 gson.toJson(
                                         new GetCiRequest(govukSigninJourneyId, ipAddress, userId)));
 
-        LOGGER.info("Retrieving CIs from CI storage system.");
+        LOGGER.info(message);
         InvokeResult result = lambdaClient.invoke(request);
 
         if (lambdaExecutionFailed(result)) {
             logLambdaExecutionError(result);
             throw new CiRetrievalException(FAILED_LAMBDA_MESSAGE);
         }
+        return result;
+    }
 
-        String jsonResponse = new String(result.getPayload().array(), StandardCharsets.UTF_8);
-        GetCiResponse response = gson.fromJson(jsonResponse, GetCiResponse.class);
-        return response.getContraIndicators();
+    private SignedJWT extractAndValidateContraIndicatorsJwt(
+            InvokeResult contraIndicatorsResult, String userId) throws CiRetrievalException {
+        final String contraIndicatorsVC =
+                new String(contraIndicatorsResult.getPayload().array(), StandardCharsets.UTF_8);
+
+        SignedJWT contraIndicatorsJwt;
+        try {
+            contraIndicatorsJwt = SignedJWT.parse(contraIndicatorsVC);
+        } catch (ParseException e) {
+            LOGGER.error(
+                    new StringMapMessage()
+                            .with(
+                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                    "Failed to parse ContraIndicators JWT.")
+                            .with(LOG_ERROR_DESCRIPTION.getFieldName(), e.getMessage()));
+            throw new CiRetrievalException("Failed to parse JWT");
+        }
+
+        final String cimitComponentId =
+                configService.getSsmParameter(ConfigurationVariable.CIMIT_COMPONENT_ID);
+        final String cimitSigningKey =
+                configService.getSsmParameter(ConfigurationVariable.CIMIT_SIGNING_KEY);
+        try {
+            verifiableCredentialJwtValidator.validate(
+                    contraIndicatorsJwt, ECKey.parse(cimitSigningKey), cimitComponentId, userId);
+        } catch (ParseException e) {
+            LOGGER.error("Error parsing CIMIT signing key: '{}'", e.getMessage());
+            throw new VerifiableCredentialException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_PARSE_CIMIT_SIGNING_KEY);
+        }
+
+        return contraIndicatorsJwt;
+    }
+
+    private ContraIndicatorEvidenceDto parseContraIndicatorEvidence(SignedJWT signedJWT)
+            throws CiRetrievalException {
+        JSONObject vcClaim;
+        try {
+            vcClaim = (JSONObject) signedJWT.getJWTClaimsSet().getClaim(VC_CLAIM);
```

Comment:
I think, can use of the signedJWT.getJWTClaimsSet().getJSONObjectClaim(VC_CLAIM);